### PR TITLE
Deglobalize Interpreter

### PR DESCRIPTION
**Pull request by AdmiralCurtiss**
Design notes: The static instruction functions remain static because the JIT (cached interpreter) calls into them, and getting pointers to member functions to call is apparently impossible in C++? Doesn't make much difference though, every one of them just gets a reference to the running Interpreter instance instead.

That aside, this is surprisingly straightforward, just a lot of `PowerPC::ppcState` -> `m_ppc_state`...

### CODE REVIEW

File: Source/Core/Core/PowerPC/JitArm64/Jit.cpp [hunk 1]
```diff
@@ -199,7 +199,8 @@ void JitArm64::FallBackToInterpreter(UGeckoInstruction inst)
 
   Interpreter::Instruction instr = Interpreter::GetInterpreterOp(inst);
   MOVP2R(ARM64Reg::X8, instr);
-  MOVI2R(ARM64Reg::W0, inst.hex);
+  MOVP2R(ARM64Reg::W0, Interpreter::getInstance());
+  MOVI2R(ARM64Reg::W1, inst.hex);
```
Comment (iwubcode) [inline Source/Core/Core/PowerPC/JitArm64/Jit.cpp:203]: Why was this added/changed?

Comment (AdmiralCurtiss) [reply Source/Core/Core/PowerPC/JitArm64/Jit.cpp:203]: `Interpreter::Instruction` was changed from `void (*)(UGeckoInstruction inst)` to `void (*)(Interpreter& interpreter, UGeckoInstruction inst)`, this adjusts the ARM JIT parameters to that.

**Review by JosJuice — APPROVED** [review]

File: Source/Core/Core/PowerPC/JitArm64/Jit.cpp (hunk 1 above)
Comment (il-av) [reply Source/Core/Core/PowerPC/JitArm64/Jit.cpp:203]: Seems like this is causing an error
MOVP2R is erroring "Can't store pointers in 32-bit registers" on Android

Comment (AdmiralCurtiss) [reply Source/Core/Core/PowerPC/JitArm64/Jit.cpp:203]: Oh right, that should be X0 I guess. I don't have a 64 bit ARM device so I'm flying kinda blind on the ARM JIT changes.

Comment (AdmiralCurtiss) [reply Source/Core/Core/PowerPC/JitArm64/Jit.cpp:203]: Does #11676 fix it?

Comment (il-av) [reply Source/Core/Core/PowerPC/JitArm64/Jit.cpp:203]: That fixes it 👍

Comment (AdmiralCurtiss) [reply Source/Core/Core/PowerPC/JitArm64/Jit.cpp:203]: Great, thanks for catching that!
